### PR TITLE
Add `align_left`/`align_right` transformers to replace `AlignLeft`/`AlignRight`

### PR DESCRIPTION
**Pull request by tanujkhattar**
- Part of #4722
- Written using the new Transformer API. 
- Supports no compile tags https://github.com/quantumlib/Cirq/issues/4253
- Default context will be supported once https://github.com/quantumlib/Cirq/issues/4879 is fixed. 
- Tests for CCO's should be added once https://github.com/quantumlib/Cirq/issues/4882 is fixed

### CODE REVIEW

File: cirq-core/cirq/transformers/align.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+from cirq.transformers import transformer_api
+
+if TYPE_CHECKING:
+    import cirq
+
+
+@transformer_api.transformer
+def align_left(
+    circuit: 'cirq.AbstractCircuit',
+    context: 'cirq.TransformerContext' = transformer_api.TransformerContext(),
```
Comment (MichaelBroughton) [inline cirq-core/cirq/transformers/align.py:28]: Nit: unsafe default.

**Review by MichaelBroughton — APPROVED** [review]
LGTM, might want to do something about the unsafe default values for context. Usually I like to do `None` followed by a promote.

File: cirq-core/cirq/transformers/align.py (hunk 1 above)
Comment (tanujkhattar) [reply cirq-core/cirq/transformers/align.py:28]: What do you mean by unsafe default?

Comment (MichaelBroughton) [reply cirq-core/cirq/transformers/align.py:28]: Having an object that might be mutable as a default argument might allow mutations to that object to persist over time which we don't want. https://google.github.io/styleguide/pyguide.html#212-default-argument-values

Comment (tanujkhattar) [reply cirq-core/cirq/transformers/align.py]: Made `context` an optional parameter with default value as None and explicit upgrade to default `TransformerContext()`.

**Pull request merged**